src/main/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlService.java

```diff
@@ -228,17 +240,30 @@ public CaseEventTrigger setReadOnlyOnCaseViewFieldsIfNoAccess(final CaseEventTri
                     CaseField field = caseFieldOpt.get();
                     if (!hasAccessControlList(userRoles, access, field.getAccessControlLists())) {
                         caseViewField.setDisplayContext(READONLY);
+                        setReadonlyOnWizardPageFieldDisplayContext(caseEventTrigger.getWizardPages(), caseViewField.getId());
                     }
                     if (field.isCompound()) {
```

Comment:
don't we need to apply the same logic for compounds fields?

Reply:
It is covered already. Please check the unit tests. For compound fields it will change the displayContext in the "complex_field_overrides": []. This one is for top level displayContext

Reply:
👍